core/arch/arm/kernel/elf_load.c

```diff
@@ -462,6 +548,23 @@ static TEE_Result e32_process_rel(struct elf_load_state *state, size_t rel_sidx,
 		case R_ARM_RELATIVE:
 			*where += vabase;
 			break;
+		case R_ARM_GLOB_DAT:
+		case R_ARM_JUMP_SLOT:
+			sym_idx = ELF32_R_SYM(rel->r_info);
+			if (sym_idx >= num_syms)
```

Comment:
Could a malformed ELF be used to abuse the TEE by out of bound indices? `sym_idx` is used to get `name_idx = mem[foo + syn_idx};` which is an index in a memory array.

Reply:
i was referring to a spectre like issue.

Reply:
Not sure... it certainly looks like a vulnerable pattern. What would be a good way to write this? Use the masking technique? (Linux has array_index_nospec() and IIRC you have proposed a similar macro in your SKS RFC PRs).

Reply:
Another option would be to include https://github.com/ARM-software/speculation-barrier/blob/master/speculation_barrier.h and use `load_no_speculate()`.